lessons/day_05/python/resolvers.py

```diff
@@ -0,0 +1,127 @@
+from ariadne import QueryType, ObjectType, EnumType, MutationType
+from random import randint
+from models import Skill, Person
+from data import session
+from datetime import datetime
+import uuid
+
+
+query = QueryType()
+mutation = MutationType()
+
+# Type definitions
+skill = ObjectType("Skill")
+person = ObjectType("Person")
+eye_color = EnumType(
+    "EyeColor",
+    {
+        'BLUE': 'blue',
+        'GREEN': 'green',
+        'BROWN': 'brown',
+        'BLACK': 'black',
+    },
+)
+
+
+# Top level resolvers
+@query.field("randomSkill")
+def resolve_random_skill(_, info):
+    records = session.query(Skill).count()
+    random_id = str(randint(1, records))
+    return session.query(Skill).get(random_id)
+
+
+@query.field("randomPerson")
+def resolve_random_person(_, info):
+    records = session.query(Person).count()
+    random_id = str(randint(1, records))
+    return session.query(Person).get(random_id)
+
+
+@query.field("person")
+def resolve_person(_, info, input=None):
+    return session.query(Person).filter_by(**input).first() if input else None
+
+
+@query.field("persons")
+def resolve_persons(_, info, input=None):
+    return session.query(Person).filter_by(**input).all() if input else session.query(Person).all()
+
+
+@query.field("skill")
+def resolve_skill(_, info, input=None):
+    return session.query(Skill).filter_by(**input).first() if input else None
+
+
+@query.field("skills")
+def resolve_skills(_, info, input=None):
+    return session.query(Skill).filter_by(**input).all() if input else session.query(Skill).all()
+
+
+# Mutations
+@mutation.field("createSkill")
+def resolve_create_skill(_, info, input):
+    skill = Skill(**input)
```

Comment:
variable name defined in outer scopes https://github.com/thinkb4/a-walk-in-graphql/blob/77841ecd9e37614f36fd488ebf49e236de1afd6a/lessons/day_06/python/resolvers.py#L39